Unnecessary 'use client' directive in Typescript declaration files

## Description

Typescript definition files generated by nuqs include `'use client'` directive, which cause typescript type error when running `tsc`. I don't see reason why this directive should be in the `*.d.ts` files. The problem can be fixed with `"skipLibCheck": true` option, but I would rather keep it as false.

## Context

package.json:
```json
{
  "name": "test",
  "version": "1.0.0",
  "main": "index.js",
  "scripts": {
    "test": "tsc"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "description": "",
  "dependencies": {
    "nuqs": "^2.8.3",
    "react": "^19.2.1",
    "typescript": "^5.9.3"
  },
  "devDependencies": {
    "@types/react": "^19.2.7"
  }
}
```

tsconfig.json:
```json
{
  "compilerOptions": {
    "target": "ESNext",
    "module": "ESNext",
    "moduleResolution": "Bundler",
    "lib": ["DOM", "DOM.Iterable", "ESNext"],
    "types": [],
    "jsx": "react",
    "strict": true,
    "noUncheckedIndexedAccess": true,
    "resolveJsonModule": true,
    "noEmit": true,
    "declaration": true,
    "declarationMap": true,
    "verbatimModuleSyntax": true,
    "isolatedDeclarations": true,
    "moduleDetection": "force",
    "downlevelIteration": true,
    "isolatedModules": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "skipLibCheck": false
  },
  "include": ["src/*.ts"],
  "exclude": ["node_modules"]
}
```

src/index.ts
```ts
import { useQueryState } from 'nuqs';
import { renderQueryString } from 'nuqs/adapters/custom';
```

## Reproduction

1. Copy the minimal setup provided in context
2. Run `npm run test`
3. See error

Gotten error:
```
$ tsc
node_modules/nuqs/dist/adapters/custom.d.ts:1:1 - error TS1036: Statements are not allowed in ambient contexts.

1 'use client';
  ~~~~~~~~~~~~

node_modules/nuqs/dist/index.d.ts:1:1 - error TS1036: Statements are not allowed in ambient contexts.

1 'use client';
  ~~~~~~~~~~~~


Found 2 errors in 2 files.

Errors  Files
     1  node_modules/nuqs/dist/adapters/custom.d.ts:1
     1  node_modules/nuqs/dist/index.d.ts:1
```

--- a/packages/nuqs/tsdown.config.ts
+++ b/packages/nuqs/tsdown.config.ts
@@ -46,7 +46,8 @@ const config: UserConfig = defineConfig([
     ...commonConfig,
     entry: entrypoints.client,
     outputOptions: {
-      intro: ({ isEntry }) => (isEntry ? "'use client';\n" : '')
+      intro: ({ isEntry, fileName }) =>
+        isEntry && !fileName.endsWith('.d.ts') ? "'use client';\n" : ''
     },
     async onSuccess() {
       // Mark the un-versionned React Router adapter as deprecated

